### ISSUE TITLE
FIFO U2F issue self hosted

### ISSUE THREAD

**Issue by derritter88** [bitwarden/web#401]
I have installed the current version of Bitwarden yesterday and paid for a premium account to use FIDO U2F.

Unfortunately it does not work on my self hosted server.
Error message: Es gab ein Problem beim lesen des Sicherheitsschlüssels, bitte erneut versuchen. 
Translation: There is an error while reading the security key, please try again.

If I try the same at vault.bitwarden.com it works without any issues.

Also I have a few other websites on my server where U2F works without a problem.

My current configuration is Apache2 as a SSL proxy + Bitwarden without SSL.

Any ideas what is not working properly?

**Comment by kspearrin** [comment]
Is the app-id.json file available? Visit https://yourdomain.com/app-id.json

**Comment by derritter88** [comment]
Hi there,

yes, the file is available:

`{
  "trustedFacets": [
    {
      "version": {
        "major": 1,
        "minor": 0
      },
      "ids": [
        "https://pw.hks-projekt.at",
        "ios:bundle-id:com.8bit.bitwarden",
        "android:apk-key-hash:dUGFzUzf3lmHSLBDBIv+WaFyZMI"
      ]
    }
  ]
}`

**Comment by derritter88** [comment]
Might it be possible that there is misconfiguration in the Apache proxy configuration?

**Comment by kspearrin** [comment]
Maybe, though it does look good to me. What service URL is configured for the web vault in ./bwdata/env/global.override.env file?

**Comment by derritter88** [comment]
they are just http + my hostname. but Apache ist dealing the SSL encryption

**Comment by kspearrin** [comment]
That's the problem. Try changing that to https. That URL is used for generating the U2F signatures, and it must be https.

**Comment by derritter88** [comment]
great, I'll try it as soon asi am at home 

________________________________
Von: Kyle Spearrin <notifications@github.com>
Gesendet: Donnerstag, 18. Juli 2019 16:23
An: bitwarden/web
Cc: derritter88; Author
Betreff: Re: [bitwarden/web] FIFO U2F issue self hosted (#401) 

That's the problem. Try changing that to https. That URL is used for generating the U2F signatures, and it must be https.

**Comment by derritter88** [comment]
It worked - thank you very much!!!
Might be helpful to add this little information to the on-premise documentation?

**Comment by KraxelHuber** [comment]
I do have exactly the same problem. I already changed all service URL entries in `./bwdata/env/global.override.env` from `http` to `https`. I then performed `./bitwarden.sh rebuild` as well as `./bitwarden.sh update`.

However, calling `https://MY_DOMAIN/app-id.json` shows `http` and not `https`. I guess, this is still causing my problem.

Any hints on how to change the settings are highly appreciated.

**Comment by derritter88** [comment]
Hi @KraxelHuber 

what's your webserver configuration?

**Comment by KraxelHuber** [comment]
I am self-hosting bitwarden on my Synology DiskStation. I am using the built-in reverse proxy.

![image](https://user-images.githubusercontent.com/29901736/91399953-2932cc00-e83f-11ea-8112-a50a6f0e473b.png)

Here's my `config.yml`:
![image](https://user-images.githubusercontent.com/29901736/91400101-65662c80-e83f-11ea-8a2d-fb7fa030475d.png)

I've got a Let's Encrypt certificate which is also handled by my DiskStation, i.e. I didn't create a certificate while setting up bitwarden on the command line.

Clearly, I replaced my real domain with `bw.MyDomain.de` ;-)
